walletlibrary/src/main/java/com/microsoft/walletlibrary/identifier/EncryptedSharedPreferencesIdentifier.kt

```diff
@@ -13,13 +14,14 @@ internal class EncryptedSharedPreferencesIdentifier(
     override val algorithm: String,
     override val method: String,
     override val keyReference: String,
-    private val cryptoOperations: CryptoOperations,
     private val keyStore: EncryptedKeyStore
 ) : HolderIdentifier {
 
-    override fun sign(data: String): String {
-        val keyId = "$id#$keyReference"
-        val privateKey = keyStore.getKey(keyReference)
-        return cryptoOperations.sign(data, privateKey, algorithm, keyId)
+    var jwsHeader: JWSHeader? = null
```

Comment:
Why is this parameter necessary? the header is passed to the ecdsaSigner, but poking into the signer it just wants to know what algorithm to use, so couldn't we construct it out of our alg parameter?